antarest/study/business/xpansion_management.py

```diff
@@ -0,0 +1,493 @@
+        self._assert_link_profile_are_files(file_study, xpansion_candidate_dto)
+        self._assert_link_exist(file_study, xpansion_candidate_dto)
+        self._assert_no_illegal_character_is_in_candidate_name(
+            xpansion_candidate_dto.name
+        )
+        if not update:
+            self._assert_candidate_name_is_not_already_taken(
+                candidates, xpansion_candidate_dto.name
+            )
+        self._assert_investment_candidate_is_valid(
+            xpansion_candidate_dto.max_investment,
+            xpansion_candidate_dto.max_units,
+            xpansion_candidate_dto.unit_size,
+        )
+
+    def add_candidate(
+        self, study: Study, xpansion_candidate_dto: XpansionCandidateDTO
+    ) -> None:
+
+        file_study = self.study_storage_service.get_storage(study).get_raw(
+            study
+        )
+
+        candidates = file_study.tree.get(["user", "expansion", "candidates"])
+
+        self._assert_candidate_is_correct(
+            candidates, file_study, xpansion_candidate_dto
+        )
+
+        # Find next candidate id
+        max_id = (
+            2 if not candidates else int(sorted(candidates.keys()).pop()) + 2
+        )
+        next_id = next(
+            str(i) for i in range(1, max_id) if str(i) not in candidates
+        )  # The primary key is actually the name, the id does not matter and is never checked.
+
+        # Add candidate
+        candidates[next_id] = xpansion_candidate_dto.dict(
+            by_alias=True, exclude_none=True
+        )
+        candidates_data = {"user": {"expansion": {"candidates": candidates}}}
+        file_study.tree.save(candidates_data)
+        # Should we add a field in the study config containing the xpansion candidates like the links or the areas ?
+
+    def get_candidate(
+        self, study: Study, candidate_name: str
+    ) -> XpansionCandidateDTO:
+        # This takes the first candidate with the given name and not the id, because the name is the primary key.
+        file_study = self.study_storage_service.get_storage(study).get_raw(
+            study
+        )
+        candidates = file_study.tree.get(["user", "expansion", "candidates"])
+        try:
+            candidate = next(
+                c for c in candidates.values() if c["name"] == candidate_name
+            )
+            return XpansionCandidateDTO(**candidate)
+
+        except StopIteration:
+            raise CandidateNotFoundError(
+                f"The candidate '{candidate_name}' does not exist"
+            )
+
+    def get_candidates(self, study: Study) -> List[XpansionCandidateDTO]:
+        file_study = self.study_storage_service.get_storage(study).get_raw(
+            study
+        )
+        candidates = file_study.tree.get(["user", "expansion", "candidates"])
+        return [XpansionCandidateDTO(**c) for c in candidates.values()]
+
+    def update_candidate(
+        self, study: Study, xpansion_candidate_dto: XpansionCandidateDTO
```

Comment:
you should add current candidate name, so you can remove it in case of renamming